code/src/pages/cart/components/CartItem.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from "react";
+
+import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
+import DeleteIcon from "@mui/icons-material/Delete";
+import RemoveCircleOutlineIcon from "@mui/icons-material/RemoveCircleOutline";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import formatPrice from "@/utils/format-price/formatPrice";
+
+import "@/pages/cart/components/CartItem.scss";
+
+type CartItemProps = {
+  item: {
+    image: string;
+    name: string;
+    productPrice: number;
+    quantity: number;
+    calculatedPrice: number;
+  };
+  onQuantityChange: (quantity: number) => void;
+  onDelete: () => void;
+};
+
+const CartItem = ({ item, onQuantityChange, onDelete }: CartItemProps) => {
+  const [quantity, setQuantity] = useState<string>(item.quantity.toString());
+
+  useEffect(() => {
+    setQuantity(item.quantity.toString());
+  }, [item.quantity]);

```

Comment:
Why we need this `useEffect`?

Reply:
> Why we need this `useEffect`?

As I talked the logic of quantity will be changed. Look at the description
It mean to setQuantity when it changed